src/samplers/gamma.jl

```diff
@@ -205,7 +205,7 @@ end
 GammaIPSampler(d::Gamma) = GammaIPSampler(d,GammaMTSampler)
 
 function rand(rng::AbstractRNG, s::GammaIPSampler)
-    x = rand(rng, s.s)
+    x = rand(rng, GammaGDSampler(s.s))
```

Comment:
Actually, I think this change is undesirable - but the current implementation is as well. The motivation for samplers is to pre-compute and pre-allocate everything required for sampling since it should pay off when you want to generate many samples. Hence by default, distributions only use dedicated samplers if multiple samples are requested (similar to the concept in Random).

So `s.s` should already be a sampler at this point here. However, it seems that's currently not the case - even though it seems that was the intention behing https://github.com/JuliaStats/Distributions.jl/blob/e67d6b3952ba775dcc10a2fc1c7546246539b10a/src/samplers/gamma.jl#L202-L205. The bug is that it does not use `S` though (and that it should be designed a bit differently probably).

Reply:
Yeah, I noticed that `S` and `GammaMTSampler` are unused. I think I found some issues on this repo from like eight years ago that mention that GammaMTSampler was "disabled".

Reply:
That comment does not refer to these lines, in any case. This bug was already present when `GammaIPSampler` was added: https://github.com/JuliaStats/Distributions.jl/pull/313

Reply:
There was a bug in the implementaiton of `GammaMTSampler`, fixing it improves performance significantly: https://github.com/JuliaStats/Distributions.jl/pull/1617#discussion_r970098297